### ISSUE TITLE
Envoy xDS RnD

### ISSUE THREAD

**Issue by mmskv** [mekstack/nataas#8]
Write a proof of concept gRPC envoy client that will add a route and TLS certificate in envoy.

- [ ] Set up local (or cloud: https://mekstack.ru, https://docs.mekstack.ru) [go-control-plane example](https://github.com/envoyproxy/go-control-plane/tree/main/internal/example) with envoy
- [ ] Try to understand how it works. You may find [envoy docs](https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/) helpful, but don't spend too much time there
- [ ] Try to configure envoy using gRPC to do reverse proxying - ideally, start more simple web services (python -m http.client) to test your envoy configuration
- [ ] Rewrite example to configure envoy your custom options which are read from config

---
This task **is** complicated, but not as much as it may seem.

What this task is not: 
- You do not need to implement protocol definitions & envoy options - they are already present in go-control-plane
- You do not need to rewrite go-control-plane

Basically, this task is about researching go-control-plane logic, retrieving needed options and then implementing a way to receive them from some source.

Note: you may skip some subtasks if you don't consider them helpful on your way to the latter two
